<!DOCTYPE html>
<html>
  <head>
    <meta charset="utf-8" >

<title>Tutorial | scMGCA</title>

<meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1, user-scalable=no">

<link rel="stylesheet" href="https://use.fontawesome.com/releases/v5.7.2/css/all.css" integrity="sha384-fnmOCqbTlWIlj8LyTjo7mOUStjsKC4pOpQbqyi7RrhN7udi9RwhKkMHpvLbHG9Sr" crossorigin="anonymous">
<link rel="shortcut icon" href="https://philyzh8.github.io/favicon.ico?v=1671523598540">
<link rel="stylesheet" href="https://philyzh8.github.io/styles/main.css">



<link rel="stylesheet" href="https://unpkg.com/aos@next/dist/aos.css" />
<script src="https://cdn.jsdelivr.net/npm/vue/dist/vue.js"></script>



    <meta name="description" content="scMGCA is a deep graph embedding learning method for single-cell clustering, which can be used to:

Single-cell data clu..." />
    <meta name="keywords" content="" />
  </head>
  <body>
    <div id="app" class="main">

      <div class="sidebar" :class="{ 'full-height': menuVisible }">
  <div class="top-container" data-aos="fade-right">
    <div class="top-header-container">
      <a class="site-title-container" href="https://philyzh8.github.io">
        <img src="https://philyzh8.github.io/images/avatar.png?v=1671523598540" class="site-logo">
        <h1 class="site-title">scMGCA</h1>
      </a>
      <div class="menu-btn" @click="menuVisible = !menuVisible">
        <div class="line"></div>
      </div>
    </div>
    <div>
      
        
          <a href="https://philyzh8.github.io/post/topological-identification-and-interpretation-for-high-throughput-single-cell-gene-regulation-elucidation-across-multiple-platforms-using-scmgca" class="site-nav">
            Home
          </a>
        
      
        
          <a href="https://philyzh8.github.io/post/installation" class="site-nav">
            Installation
          </a>
        
      
        
          <a href="https://philyzh8.github.io/post/tutorial" class="site-nav">
            Tutorial
          </a>
        
      
    </div>
  </div>
  <div class="bottom-container" data-aos="flip-up" data-aos-offset="0">
    <div class="social-container">
      
        
      
        
      
        
      
        
      
        
      
    </div>
    <div class="site-description">
      
    </div>
    <div class="site-footer">
      <a href="https://github.com/Philyzh8/scMGCA">scMGCA</a> | <a class="rss" href="https://philyzh8.github.io/atom.xml" target="_blank">RSS</a>
    </div>
  </div>
</div>


      <div class="main-container">
        <div class="content-container" data-aos="fade-up">
          <div class="post-detail">
            <h2 class="post-title">Tutorial</h2>
            <div class="post-date">2022-11-04</div>
            
            <div class="post-content" v-pre>
              <p><code>scMGCA</code> is a deep graph embedding learning method for single-cell clustering, which can be used to:</p>
<ul>
<li>Single-cell data clustering. The example can be seen in the <a href="https://github.com/Philyzh8/scMGCA/blob/master/tutorial/demo.py">demo.py</a>.</li>
<li>Correct the batch effect of data from different scRNA-seq protocols. The example can be seen in the <a href="https://github.com/Philyzh8/scMGCA/blob/master/tutorial/demo_batch.py">demo_batch.py</a>.</li>
<li>Analysis of the mouse brain data with 1.3 million cells. The example can be seen in the <a href="https://github.com/Philyzh8/scMGCA/blob/master/tutorial/demo_scale.py">demo_scale.py</a>.</li>
<li>Provide a automatic supershell search algorithm. The example can be seen in the <a href="https://github.com/Philyzh8/scMGCA/blob/master/tutorial/demo_para.py">demo_para.py</a>.</li>
</ul>
<p>We give users some suggestions for running in the <a href="https://github.com/Philyzh8/scMGCA/blob/master/tutorial/tutorial.md">tutorial.md</a>.</p>
<p>In this tutorial, we will perform the entire <code>scMGCA</code> cluster analysis using the Qx Bladder dataset dataset (can be downloaded <a href="https://drive.google.com/drive/folders/1BIZxZNbouPtGf_cyu7vM44G5EcbxECeu">here</a>), and the entire <code>scMGCA</code> batch effect correction using the human pancreatic dataset (can be downloaded <a href="https://github.com/Philyzh8/scMGCA/tree/master/dataset/pancreas">here</a>).</p>
<h2 id="cluster-analysis">Cluster analysis</h2>
<h3 id="import-python-package">Import python package</h3>
<p>Import the <code>scMGCA</code>  module for cluster analysis and other python packages.</p>
<pre><code>import os
import argparse
import pandas as pd
import tensorflow as tf
from spektral.layers import GraphConv
from sklearn import metrics
from numpy.random import seed
seed(1)
tf.random.set_seed(1)
# Remove warnings
import warnings
warnings.simplefilter(action='ignore', category=FutureWarning)
tf.compat.v1.logging.set_verbosity(tf.compat.v1.logging.ERROR)
# scMGCA module
from scMGCA.preprocess import *
from scMGCA.utils import *
from scMGCA.scmgca import SCMGCA, SCMGCAL
from scMGCA.losses import *
from scMGCA.graph_function import *
</code></pre>
<h3 id="parameter-settings">Parameter settings</h3>
<p>Set some parameters, including selecting the number of highly variable genes, pre-training epochs and training epochs.</p>
<pre><code>parser = argparse.ArgumentParser(description=&quot;train&quot;, formatter_class=argparse.ArgumentDefaultsHelpFormatter)
parser.add_argument(&quot;--dataname&quot;, default = &quot;Quake_10x_Bladder&quot;, type = str)
parser.add_argument(&quot;--highly_genes&quot;, default = 500, type=int)
parser.add_argument(&quot;--pretrain_epochs&quot;, default = 1000, type=int)
parser.add_argument(&quot;--maxiter&quot;, default = 300, type=int)
args = parser.parse_args()
</code></pre>
<h3 id="read-data">Read data</h3>
<p>Read raw gene expression data and true labels from <code>.h5</code> files.</p>
<pre><code>data = './dataset/'+args.dataname+'/data.h5'
x, y = prepro(data)
cluster_number = len(np.unique(y))
print(&quot;Cell number:&quot;, x.shape[0])
print(&quot;Gene number&quot;,x.shape[1])        
print(&quot;Cluster number:&quot;, cluster_number)
</code></pre>
<h3 id="data-preprocessing">Data preprocessing</h3>
<p><code>scMGCA</code> uses <code>AnnData</code> object data to preserve data, and uses <code>SCANPY</code> to preprocess the data, including cells and genes filtering, normalization, logarithm transformation, Selection of highly variable genes and scaling.</p>
<pre><code>x = np.ceil(x).astype(np.int)
adata = sc.AnnData(x)
adata.obs['Group'] = y
adata = normalize(adata, copy=True, highly_genes=args.highly_genes, size_factors=True, normalize_input=True, logtrans_input=True)
</code></pre>
<h3 id="construct-cell-graph">Construct cell graph</h3>
<p>Construct a cell-PPMI graph and use it as an input graph in GCN.</p>
<pre><code>count = adata.X
adj = get_adj(count)
adj_n = GraphConv.preprocess(adj)
</code></pre>
<h3 id="pre-training">Pre-training</h3>
<p>Build the <code>scMGCA</code> model and perform pre-training to obtain the low-dimensional representation.</p>
<pre><code>model = SCMGCA(count, adj=adj, adj_n=adj_n)
model.pre_train(epochs=args.pretrain_epochs)
latent_pre = model.embedding(count, adj_n)
</code></pre>
<h3 id="training">Training</h3>
<p>The cluster centers are initialized on the pretrained low-dimensional representation and then input into the <code>scMGCA</code> cluster analysis model for training.</p>
<pre><code>centers = init_center(args, latent_pre, adj_n, cluster_number)
Cluster_predicted=model.train(epochs=args.maxiter, centers=centers)
</code></pre>
<h3 id="evaluation">Evaluation</h3>
<p>Evaluate the clustering results</p>
<pre><code>y = list(map(int, y))
Cluster_predicted.y_pred = np.array(Cluster_predicted.y_pred)
nmi = metrics.normalized_mutual_info_score(y, Cluster_predicted.y_pred)
ari = metrics.adjusted_rand_score(y, Cluster_predicted.y_pred)
print('NMI= %.4f, ARI= %.4f' % (nmi, ari))
</code></pre>
<pre><code>NMI= 0.9802, ARI= 0.9900
</code></pre>
<h2 id="batch-effect-correction">Batch effect correction</h2>
<h3 id="import-python-package-2">Import python package</h3>
<p>Import the <code>scMGCA</code>  module for batch effect correction and other python packages.</p>
<pre><code>import os
import argparse
import pandas as pd
import tensorflow as tf
from spektral.layers import GraphConv
from sklearn import metrics
from numpy.random import seed
seed(1)
tf.random.set_seed(1)
# Remove warnings
import warnings
warnings.simplefilter(action='ignore', category=FutureWarning)
tf.compat.v1.logging.set_verbosity(tf.compat.v1.logging.ERROR)
# scMGCA module
from scMGCA.preprocess import *
from scMGCA.utils import *
from scMGCA.scmgca import SCMGCA, SCMGCAL
from scMGCA.losses import *
from scMGCA.graph_function import *
</code></pre>
<h3 id="parameter-settings-2">Parameter settings</h3>
<p>Set some parameters, including selecting the number of highly variable genes, pre-training epochs and training epochs.</p>
<pre><code>parser = argparse.ArgumentParser(description=&quot;train&quot;, formatter_class=argparse.ArgumentDefaultsHelpFormatter)
parser.add_argument(&quot;--dataname&quot;, default = &quot;pancreas&quot;, type = str)
parser.add_argument(&quot;--highly_genes&quot;, default = 2000, type=int)
parser.add_argument(&quot;--pretrain_epochs&quot;, default = 1000, type=int)
parser.add_argument(&quot;--maxiter&quot;, default = 300, type=int)
args = parser.parse_args()
</code></pre>
<h3 id="read-data-2">Read data</h3>
<p>Read raw gene expression data and true labels from <code>.h5</code> files.</p>
<pre><code>adata = read_pancreas(&quot;./dataset/pancreas&quot;, cache=True)
y = np.array(adata.obs['celltype'].values, dtype=str)
cluster_number = len(np.unique(y))
print(&quot;Cell number:&quot;, adata.X.shape[0])
print(&quot;Gene number&quot;,adata.X.shape[1])   
print(&quot;Cluster number:&quot;, cluster_number)
</code></pre>
<h3 id="data-preprocessing-2">Data preprocessing</h3>
<p><code>scMGCA</code> uses <code>AnnData</code> object data to preserve data, and uses <code>SCANPY</code> to preprocess the data, including cells and genes filtering, normalization, logarithm transformation, Selection of highly variable genes and scaling.</p>
<pre><code>count = normalize_batch(adata, batch_key = 'tech', n_high_var = args.highly_genes)  
print(count.shape)
</code></pre>
<h3 id="construct-cell-graph-2">Construct cell graph</h3>
<p>Construct a cell-PPMI graph and use it as an input graph in GCN.</p>
<pre><code>adj = get_adj(count)
adj_n = GraphConv.preprocess(adj)
</code></pre>
<h3 id="pre-training-2">Pre-training</h3>
<p>Build the <code>scMGCA</code> model and perform pre-training to obtain the low-dimensional representation.</p>
<pre><code>model = SCMGCA(count, adj=adj, adj_n=adj_n, latent_dim=20, dec_dim=[128])
model.pre_train(epochs=args.pretrain_epochs, W_a=0.6, lr=5e-4)
latent_pre = model.embedding(count, adj_n)
</code></pre>
<h3 id="training-2">Training</h3>
<p>The cluster centers are initialized on the pretrained low-dimensional representation and then input into the <code>scMGCA</code> cluster analysis model for training.</p>
<pre><code>adata_latent = sc.AnnData(latent_pre)
sc.pp.neighbors(adata_latent, n_neighbors = 15, use_rep=&quot;X&quot;)
resolution = find_resolution(adata_latent, 8, 0)
adata_latent = sc.tl.leiden(adata_latent, resolution = resolution, random_state = 0, copy = True)
Y_pred_init = np.asarray(adata_latent.obs['leiden'], dtype=int)
features = pd.DataFrame(adata_latent.X, index = np.arange(0, adata_latent.shape[0]))
Group = pd.Series(Y_pred_init, index = np.arange(0, adata_latent.shape[0]), name=&quot;Group&quot;)
Mergefeature = pd.concat([features, Group],axis=1)
centers = np.asarray(Mergefeature.groupby(&quot;Group&quot;).mean())
Cluster_predicted=model.train(epochs=args.maxiter, W_a=0.6, centers=centers)
</code></pre>
<h3 id="evaluation-2">Evaluation</h3>
<p>Evaluate the clustering results</p>
<pre><code>y = list(map(str, y))
Cluster_predicted.y_pred = np.array(Cluster_predicted.y_pred)
nmi = metrics.normalized_mutual_info_score(y, Cluster_predicted.y_pred)
ari = metrics.adjusted_rand_score(y, Cluster_predicted.y_pred)
print('NMI= %.4f, ARI= %.4f' % (nmi, ari))
</code></pre>
<pre><code>NMI= 0.9275, ARI= 0.9585
</code></pre>
<h3 id="visualization">Visualization</h3>
<p>Visualize latent representations via UMAP and label them with ground-truth batch labels.</p>
<pre><code>import matplotlib.pyplot as plt
import umap
latent = Cluster_predicted.latent
reducer = umap.UMAP(n_neighbors=20,metric='cosine',min_dist=0.01,random_state=42)
embedding = reducer.fit_transform(latent)
for i, x in enumerate(adata.obs['tech']):
    if x == 'celseq':
        s1 = plt.scatter(embedding[i][0], embedding[i][1], s=5, color='#019013')
    elif x == 'celseq2':
        s2 = plt.scatter(embedding[i][0], embedding[i][1], s=5, color='#FD8400')
    elif x == 'fluidigmc1':
        s3 = plt.scatter(embedding[i][0], embedding[i][1], s=5, color='#BD9001')
    elif x == 'smartseq2':
        s4 = plt.scatter(embedding[i][0], embedding[i][1], s=5, color='#00BAAC')

plt.xticks([])
plt.yticks([])
plt.xlabel('UMAP1', fontsize=25)
plt.ylabel('UAMP2', fontsize=25)

ax = plt.gca()
ax.spines['right'].set_color('black')
ax.spines['top'].set_color('black')
ax.spines['left'].set_color('black')
ax.spines['bottom'].set_color('black')
ax.spines['bottom'].set_linewidth(2)
ax.spines['left'].set_linewidth(2)
ax.spines['right'].set_linewidth(2)
ax.spines['top'].set_linewidth(2)  
plt.show()
</code></pre>
<figure data-type="image" tabindex="1"><img src="https://philyzh8.github.io/post-images/1668403944135.jpg" alt="" loading="lazy"></figure>

            </div>
            
            
              <div class="next-post">
                <div class="next">下一篇</div>
                <a href="https://philyzh8.github.io/post/installation/">
                  <h3 class="post-title">
                    Installation
                  </h3>
                </a>
              </div>
            

            

          </div>

        </div>
      </div>
    </div>

    <script src="https://unpkg.com/aos@next/dist/aos.js"></script>
<script type="application/javascript">

AOS.init();

var app = new Vue({
  el: '#app',
  data: {
    menuVisible: false,
  },
})

</script>


  <script src="//cdn.jsdelivr.net/gh/highlightjs/cdn-release@11.5.1/build/highlight.min.js"></script>
  <script>
    hljs.initHighlightingOnLoad()
  </script>





  </body>
</html>
