<!DOCTYPE html>
<html>
  <head>
    <meta charset="utf-8" >

<title>üìãHomework | scMGCA</title>

<meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1, user-scalable=no">

<link rel="stylesheet" href="https://use.fontawesome.com/releases/v5.7.2/css/all.css" integrity="sha384-fnmOCqbTlWIlj8LyTjo7mOUStjsKC4pOpQbqyi7RrhN7udi9RwhKkMHpvLbHG9Sr" crossorigin="anonymous">
<link rel="shortcut icon" href="https://philyzh8.github.io/favicon.ico?v=1671523598540">
<link rel="stylesheet" href="https://philyzh8.github.io/styles/main.css">



<link rel="stylesheet" href="https://unpkg.com/aos@next/dist/aos.css" />
<script src="https://cdn.jsdelivr.net/npm/vue/dist/vue.js"></script>



    <meta name="description" content="üè∑Ô∏è Scanpy is a scalable toolkit (Python-based package) for analyzing single-cell gene expression data built jointly wit..." />
    <meta name="keywords" content="" />
  </head>
  <body>
    <div id="app" class="main">

      <div class="sidebar" :class="{ 'full-height': menuVisible }">
  <div class="top-container" data-aos="fade-right">
    <div class="top-header-container">
      <a class="site-title-container" href="https://philyzh8.github.io">
        <img src="https://philyzh8.github.io/images/avatar.png?v=1671523598540" class="site-logo">
        <h1 class="site-title">scMGCA</h1>
      </a>
      <div class="menu-btn" @click="menuVisible = !menuVisible">
        <div class="line"></div>
      </div>
    </div>
    <div>
      
        
          <a href="https://philyzh8.github.io/post/topological-identification-and-interpretation-for-high-throughput-single-cell-gene-regulation-elucidation-across-multiple-platforms-using-scmgca" class="site-nav">
            Home
          </a>
        
      
        
          <a href="https://philyzh8.github.io/post/installation" class="site-nav">
            Installation
          </a>
        
      
        
          <a href="https://philyzh8.github.io/post/tutorial" class="site-nav">
            Tutorial
          </a>
        
      
    </div>
  </div>
  <div class="bottom-container" data-aos="flip-up" data-aos-offset="0">
    <div class="social-container">
      
        
      
        
      
        
      
        
      
        
      
    </div>
    <div class="site-description">
      
    </div>
    <div class="site-footer">
      <a href="https://github.com/Philyzh8/scMGCA">scMGCA</a> | <a class="rss" href="https://philyzh8.github.io/atom.xml" target="_blank">RSS</a>
    </div>
  </div>
</div>


      <div class="main-container">
        <div class="content-container" data-aos="fade-up">
          <div class="post-detail">
            <h2 class="post-title">üìãHomework</h2>
            <div class="post-date">2022-09-29</div>
            
            <div class="post-content" v-pre>
              <p>üè∑Ô∏è Scanpy is a scalable toolkit (Python-based package) for analyzing single-cell gene expression data built jointly with anndata. It includes preprocessing, visualization, clustering, trajectory inference and differential expression testing. Please use Scanpy to preprocess (including filtering, quality control, normalization, highly variable gene selection) and cluster the <a href="https://cf.10xgenomics.com/samples/cell-exp/1.1.0/pbmc3k/pbmc3k_filtered_gene_bc_matrices.tar.gz"><strong>data</strong></a>, and submit the visualization of the clustering results</p>
<p>üè∑Ô∏èSeurat is an R package designed for QC (quality control), analysis, and exploration of single-cell RNA-seq data.  Seurat aims to enable users to identify and interpret sources of heterogeneity from single-cell transcriptomic measurements, and to integrate diverse types of single-cell data. Please use Seurat to preprocess (including quality control, normalization, highly variable gene selection, scale, dimensional reduction) and cluster the <a href="https://cf.10xgenomics.com/samples/cell-exp/1.1.0/pbmc3k/pbmc3k_filtered_gene_bc_matrices.tar.gz"><strong>data</strong></a>. Then find differentially expressed genes (i.e. features) based on the clustering results and visualize the differentially expressed genes with the highest expression levels in each cluster (violin plot).</p>

            </div>
            
            

            

          </div>

        </div>
      </div>
    </div>

    <script src="https://unpkg.com/aos@next/dist/aos.js"></script>
<script type="application/javascript">

AOS.init();

var app = new Vue({
  el: '#app',
  data: {
    menuVisible: false,
  },
})

</script>


  <script src="//cdn.jsdelivr.net/gh/highlightjs/cdn-release@11.5.1/build/highlight.min.js"></script>
  <script>
    hljs.initHighlightingOnLoad()
  </script>





  </body>
</html>
